torch/utils/launch_cpu_inference.py

```diff
@@ -0,0 +1,577 @@
+            if item.endswith("lib{lib_type}.so"):
+                lib_set = True
+                break
+        if not lib_set:
+            for lib_path in library_paths:
+                library_file = f"{lib_path}/lib{lib_type}.so"
+                matches = glob.glob(library_file)
+                if len(matches) > 0:
+                    if "LD_PRELOAD" in os.environ:
+                        os.environ["LD_PRELOAD"] = f"{matches[0]}:{os.environ['LD_PRELOAD']}"
+                    else:
+                        os.environ["LD_PRELOAD"] = matches[0]
+                    lib_find = True
+                    break
+        return lib_set or lib_find
+
+
+    def set_memory_allocator(self, enable_tcmalloc=True, enable_jemalloc=False, use_default_allocator=False):
+        """
+        Enable TCMalloc/JeMalloc with LD_PRELOAD and set configuration for JeMalloc.
+        By default, PTMalloc will be used for PyTorch, but TCMalloc and JeMalloc can get better
+        memory resue and reduce page fault to improve performance.
+        """
+        if enable_tcmalloc and enable_jemalloc:
+            logger.error("Unable to enable TCMalloc and JEMalloc at the same time")
+            exit(-1)
+
+        if enable_tcmalloc:
+            find_tc = self.add_lib_preload(lib_type="tcmalloc")
+            if not find_tc:
+                msg = f"{msg_log_notfound} you can use \"conda install -c conda-forge gperftools\" to install {{0}}"
+                logger.warning(msg.format("TCmalloc", "tcmalloc"))
+            else:
+                logger.info("Use TCMalloc memory allocator")
+
+        elif enable_jemalloc:
+            find_je = self.add_lib_preload(lib_type="jemalloc")
+            if not find_je:
+                msg = f"{msg_log_notfound} you can use \"conda install -c conda-forge jemalloc\" to install {{0}}"
+                logger.warning(msg.format("Jemalloc", "jemalloc"))
+            else:
+                logger.info("Use JeMalloc memory allocator")
+                self.set_env("MALLOC_CONF", "oversize_threshold:1,background_thread:true,metadata_thp:auto")
+
+        elif use_default_allocator:
+            pass
+
+        else:
+            find_tc = self.add_lib_preload(lib_type="tcmalloc")
+            if find_tc:
+                logger.info("Use TCMalloc memory allocator")
+                return
+            find_je = self.add_lib_preload(lib_type="jemalloc")
+            if find_je:
+                logger.info("Use JeMalloc memory allocator")
+                return
+            logger.warning(f"""Neither TCMalloc nor JeMalloc is found in $CONDA_PREFIX/lib or $VIRTUAL_ENV/lib
+                            or /.local/lib/ or /usr/local/lib/ or /usr/local/lib64/ or /usr/lib or /usr/lib64 or
+                           {expanduser("~")}/.local/lib/ so the LD_PRELOAD environment variable will not be set. This may drop the performance""")
+
+    def logger_env(self, env_name=""):
```

Comment:
this method logs the environment variable name and value correct? then it should be called `log_env_var(env_var_name)`

Reply:
OK. This prints an environment variable with name and value pair.